nebula_ros/include/nebula_ros/continental/continental_ars548_hw_interface_ros_wrapper.hpp

```diff
@@ -0,0 +1,140 @@
+#include "nebula_ros/common/nebula_hw_interface_ros_wrapper_base.hpp"
+
+#include <ament_index_cpp/get_package_prefix.hpp>
+#include <diagnostic_updater/diagnostic_updater.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp_components/register_node_macro.hpp>
+
+#include "nebula_msgs/msg/nebula_packet.hpp"
+#include "nebula_msgs/msg/nebula_packets.hpp"
+#include "std_msgs/msg/bool.hpp"
+#include "std_srvs/srv/empty.hpp"
+
+#include <boost/asio.hpp>
+
+#include <memory>
+#include <mutex>
+#include <string>
+#include <thread>
+#include <vector>
+
+namespace nebula
+{
+namespace ros
+{
+/// @brief Get parameter from rclcpp::Parameter
+/// @tparam T
+/// @param p Parameter from rclcpp parameter callback
+/// @param name Target parameter name
+/// @param value Corresponding value
+/// @return Whether the target name existed
+template <typename T>
+bool get_param(const std::vector<rclcpp::Parameter> & p, const std::string & name, T & value)
+{
+  auto it = std::find_if(p.cbegin(), p.cend(), [&name](const rclcpp::Parameter & parameter) {
+    return parameter.get_name() == name;
+  });
+  if (it != p.cend()) {
+    value = it->template get_value<T>();
+    return true;
+  }
+  return false;
+}
+
+/// @brief Hardware interface ros wrapper of continental radar ethernet driver
+class ContinentalARS548HwInterfaceRosWrapper final : public rclcpp::Node,
+                                                     NebulaHwInterfaceWrapperBase
+{
+  drivers::continental_ars548::ContinentalARS548HwInterface hw_interface_;
+  Status interface_status_;
+
+  drivers::ContinentalARS548SensorConfiguration sensor_configuration_;
+
+  /// @brief Received Continental Radar message publisher
+  rclcpp::Publisher<nebula_msgs::msg::NebulaPackets>::SharedPtr packets_pub_;
+  rclcpp::Subscription<std_msgs::msg::Bool>::SharedPtr driving_direction_sub_;
+  rclcpp::Service<std_srvs::srv::Empty>::SharedPtr sensor_config_service_server_;
+
+  /// @brief Initializing hardware interface ros wrapper
+  /// @param sensor_configuration SensorConfiguration for this driver
+  /// @return Resulting status
+  Status InitializeHwInterface(
+    const drivers::SensorConfigurationBase & sensor_configuration) override;
+  /// @brief Callback for receiving NebulaPackets
+  /// @param packets_buffer Received NebulaPackets
+  void ReceivePacketsDataCallback(std::unique_ptr<nebula_msgs::msg::NebulaPackets> packets_buffer);
+
+  /// @brief (Dummy) Callback to set the driving direction
+  /// @param msg A dummy msg
+  void DrivingDirectionCallback(const std_msgs::msg::Bool::SharedPtr msg);
+
+  /// @brief (Dummy) Service callback to configure the sensor
+  /// @param request Empty service request
+  /// @param response Empty service response
+  void SensorConfigureRequestCallback(
+    const std::shared_ptr<std_srvs::srv::Empty::Request> request,
+    const std::shared_ptr<std_srvs::srv::Empty::Response> response);
```

Comment:
Please provide an explanation (e.g. not yet implemented, needed for XYZ) on why these are dummy callbacks. As it is now, it is not immediately clear to the reader why they exist at all.

Reply:
@mojomex 
With respect to the comments in the ros_wrappers, how to use them both generally in ros or autoware has not been discussed yet (hence the reason this PR is still a draft).

I will address this comments later when we have a better idea of how to handle all of our use-cases